test/driver_stub/cl.c

```diff
@@ -355,6 +355,30 @@ clGetContextInfo(cl_context         context,
     return return_value;
 }
 
+#ifdef CL_VERSION_3_0
+
+CL_API_ENTRY cl_int CL_API_CALL
+clSetContextDestructorCallback(cl_context         context,
+                               void (CL_CALLBACK* pfn_notify)(cl_context context,
+                                                              void* user_data),
+                               void*              user_data) CL_API_SUFFIX__VERSION_3_0
+{
+    cl_int return_value = CL_OUT_OF_RESOURCES;
+    test_icd_stub_log("clSetContextDestructorCallback(%p, %p, %p)\n",
+                      context,
+                      pfn_notify,
+                      user_data);
+    pfn_notify(context, NULL);
+    test_icd_stub_log("setcontextdestructor_callback(%p, %p)\n",
+               context,
+               NULL);
```

Comment:
```suggestion
    pfn_notify(context, user_data);
    test_icd_stub_log("setcontextdestructor_callback(%p, %p)\n",
               context,
               user_data);
```

I didn't actually try this out.

It doesn't make any difference to the tests, so not too important whether the stub does this or not.

Reply:
Interesting - it looks like all of the tests pass `NULL` unconditionally to the callbacks vs. `user_data`.

I've fixed this for both `clSetMemObjectDestructorCallback` and `clSetContextDestructorCallback` with the change suggested above. I didn't touch the other callbacks for now though, and if we want to fix these I think we should do it as a follow-on PR.

Thanks!

Reply:
Looks good, thanks!